### PR TITLE
[batch] simplify batch api.py

### PR DESCRIPTION
**Pull request by danking**
Get rid of default API and all these unnecessary api root-level methods. Add a cookies and headers parameter that can be used to send arbitrary cookies and headers along.

### CODE REVIEW

File: batch/batch/api.py [hunk 1]
```diff
@@ -37,34 +68,22 @@ def create_job(self, url, spec, attributes, batch_id, callback, parent_ids,
         if copy_service_account_name:
             doc['copy_service_account_name'] = copy_service_account_name
 
-        response = requests.post(url + '/jobs/create', json=doc, timeout=self.timeout)
-        raise_on_failure(response)
-        return response.json()
+        return self.post(url + '/jobs/create', json=doc)
```
Comment (akotlar) [inline batch/batch/api.py]: Could we use 1 f string for all of these?

File: batch/batch/api.py [hunk 2]
```diff
@@ -74,76 +93,13 @@ def create_batch(self, url, attributes, callback, ttl):
...
-        raise_on_failure(response)
-        return response.json()
+        return self.post(url + f'/batches/{batch_id}/close')
 
     def delete_batch(self, url, batch_id):
-        response = requests.delete(url + '/batches/{}'.format(batch_id), timeout=self.timeout)
-        raise_on_failure(response)
-        return response.json()
-
-    def refresh_k8s_state(self, url):
```
Comment (akotlar) [inline batch/batch/api.py]: Want to check that this Api method was meant to go

Comment (akotlar) [reply batch/batch/api.py]: If so, https://github.com/hail-is/hail/blob/80d7c9020b99fc446640abc2c2f2068669347d71/batch/batch/client.py#L209 and the corresponding client test need to be removed.

File: batch/batch/api.py [hunk 3]
```diff
@@ -74,76 +93,16 @@ def create_batch(self, url, attributes, callback, ttl):
...
-    return DEFAULT_API.close_batch(url, batch_id)
-
-
-def delete_batch(url, batch_id):
-    return DEFAULT_API.delete_batch(url, batch_id)
-
-
-def refresh_k8s_state(url):
-    return DEFAULT_API.refresh_k8s_state(url)
+        return self.post(f'{url}/refresh_k8s_state')
```
Comment (akotlar) [inline batch/batch/api.py]: Used to return None now returns json

**Comment by cseed** [comment]
This is approved but needs rebase.  I'm making some more API changes.  If you can rebase soon, I will wait until this goes in to avoid further conflicts.

File: ci/ci/environment.py [hunk 4]
```diff
@@ -59,4 +59,4 @@
 log.info(f'CONTEXT = {CONTEXT}')
 
 
-batch_client = BatchClient(url=BATCH_SERVER_URL, api=API(timeout=5))
+batch_client = BatchClient(url=BATCH_SERVER_URL, timeout=5)
```
Comment (cseed) [inline ci/ci/environment.py]: This is failing, BatchClient has no timeout argument.

**Pull request merged**